### PR TITLE
[Reusable Component] Pagination

### PR DESCRIPTION
**Pull request by mikorosales**
## Issue Link
[LMS-315](https://framgiaph.backlog.com/view/LMS-315)

## Defintion of Done
- Able to display a functional pagination component in this [url ](http://localhost:3000/demo/components#pagination)
- Page number should also display in the url upon user interaction

## Steps to reproduce
- Import Pagination in the page you want it to be utilized

## Affected Components / Functionalities / Page
n/a

## Test Cases
_will update soon..._

## Notes
- To properly demonstrate page event handling you can add this code as well in the page
```
  // sets up the state of the page to track user interaction
  const [currentPage, setCurrentPage] = useState(2);
  const handleChangePageEvent = (page: number): void => {
    setCurrentPage(page);
  };
```

## Screenshots
image
image

### CODE REVIEW

File: client/src/shared/components/Pagination/index.tsx [hunk 1]
```diff
@@ -0,0 +1,98 @@
+import React from 'react';
+import Link from 'next/link';
+import ChevronLeft from '../../icons/ChevronLeft';
+import ChevronRight from '../../icons/ChevronRight';
```
Comment (kdacudag) [inline client/src/shared/components/Pagination/index.tsx:4]: Use the absolute path `@/src/shared` instead of relative path

File: client/src/shared/components/Pagination/index.tsx [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+      </ul>
+    </nav>
+  );
+};
+
+Pagination.defaultProps = {
+  maxPages: 5,
+  totalPages: 10,
+  currentPage: 1
+};
```
Comment (escbooster12-dev) [inline client/src/shared/components/Pagination/index.tsx:96]: - these props should be required. So please do not define default props on it.

File: client/src/shared/components/Pagination/index.tsx [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+      onChangePage(page);
+    }
+  };
+
+  return (
+    <nav>
+      <ul className="flex">
+        {!isFirstPage && (
+          <li className="mr-1">
+            <Link href={`/demo/components/?page=${prevPage}`}>
```
Comment (escbooster12-dev) [inline client/src/shared/components/Pagination/index.tsx:38]: - do not render a link for this. we avoid jumping the page from to another.
- let the handler do what he wants to do. (if he wants to jump the page or not)
- for demonstration purposes okay not to jump from one page to another

File: client/src/shared/icons/ChevronLeft.tsx [hunk 4]
```diff
@@ -0,0 +1,22 @@
+import React from 'react';
+
+const ChevronLeft: React.FC = () => {
```
Comment (escbooster12-dev) [inline client/src/shared/icons/ChevronLeft.tsx:3]: - please add the necessary props. (you can check existing icons for reference).

File: client/styles/globals.css [hunk 5]
```diff
@@ -1,3 +1,4 @@
 @tailwind base;
 @tailwind components;
 @tailwind utilities;
+
```
Comment (escbooster12-dev) [inline client/styles/globals.css:4]: - unnecessary changes

**Review by Marielleregner7 — APPROVED** [review]
**Approved**
- functional pagination component is visible in this [url](http://localhost:3000/demo/components#pagination)
- Page number is also displayed in the url upon user interaction
- No ESLint warnings or errors.

Comments: 6
Test time: 18 mins
Iterations: 1

**Review by escbooster12-dev — COMMENTED** [review]
Comments: 5
Iteration: 2
